### PR TITLE
OWPaintData: PyLint

### PR DESCRIPTION
**Pull request by janezd**
Pylint reported a lot of irrelevant issues for Paint Data, so this PR is mostly about disabling them.

### CODE REVIEW

File: Orange/widgets/data/owpaintdata.py [hunk 1]
```diff
@@ -594,6 +596,8 @@ def undo(self):
         self._model.execute(self._undo)
 
     def mergeWith(self, other):
+        # other is another instance of the same class, thus
+        # pylint: disable=protected-access
         if self.id() != other.id():
```
Comment (janezd) [inline Orange/widgets/data/owpaintdata.py]: @ales-erjavec, isn't `id` always `1`? If I'm not mistaken, this check, the above comment, and method `id` can be removed?

File: Orange/widgets/data/owpaintdata.py [hunk 2]
```diff
@@ -1154,20 +1158,20 @@ def _on_editing_finished(self):
...
     def execute(self, command):
-        if isinstance(command, (Append, DeleteIndices, Insert, Move)):
-            if isinstance(command, (DeleteIndices, Insert)):
-                self._selected_indices = None
+        assert isinstance(command, (Append, DeleteIndices, Insert, Move)), \
+            "Non normalized command"
+        if isinstance(command, (DeleteIndices, Insert)):
+            self._selected_indices = None
 
-                if isinstance(self.current_tool, SelectTool):
```
Comment (janezd) [inline Orange/widgets/data/owpaintdata.py]: @ales-erjavec, should this `if` really be nested? Am I overlooking something or is this condition always false?

If resetting the tool is redundant, `_reset()` method can remain private.

File: Orange/widgets/data/owpaintdata.py (hunk 1 above)
Comment (ales-erjavec) [reply Orange/widgets/data/owpaintdata.py]: The check can be, but the `id` method [should stay.](https://doc.qt.io/qt-5/qundocommand.html#id)

File: Orange/widgets/data/owpaintdata.py (hunk 2 above)
Comment (ales-erjavec) [reply Orange/widgets/data/owpaintdata.py]: Yes the condition can be true and should be nested.

Comment (janezd) [reply Orange/widgets/data/owpaintdata.py]: Oops, apologies for asking. I was so sloppy that I overlooked that this one checks `self.current_tool`, not `command`, so I didn't understand why is this needed. :(

**Pull request merged**